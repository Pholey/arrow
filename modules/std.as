import "libc";

// Exit the program (with the given status)
def exit(status: int32) {
  libc.exit(libc.c_int(status));
}

// Check the condition and abort program execution if false
def assert(condition: bool) {
  if not condition {
    libc.abort();
  }
}

// Returns the size of a type in bytes
def size_of<T>(): uint64 {
  return (((0 as *T) + 1) as *uint8) - ((0 as *T) as *uint8);
}
