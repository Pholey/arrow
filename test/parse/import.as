import "os";
import "std";
import "@mehcode/some_cool_lib";
import "R*@ !) F" as ra;
