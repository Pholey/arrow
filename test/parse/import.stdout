{
  "tag": "Module",
  "span": "test/parse/import.as:1-4",
  "statements": [
    {
      "tag": "Import",
      "span": "test/parse/import.as:1:1-12",
      "source": "os",
      "alias": ""
    },
    {
      "tag": "Import",
      "span": "test/parse/import.as:2:1-13",
      "source": "std",
      "alias": ""
    },
    {
      "tag": "Import",
      "span": "test/parse/import.as:3:1-32",
      "source": "@mehcode/some_cool_lib",
      "alias": ""
    },
    {
      "tag": "Import",
      "span": "test/parse/import.as:4:1-24",
      "source": "R*@ !) F",
      "alias": "ra"
    }
  ]
}
